src/EA.Weee.Web/Areas/AATFReturn/Controllers/ReceivedPCSListController.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace EA.Weee.Web.Areas.AatfReturn.Controllers
+{
+    using Api.Client;
+    using Constant;
+    using EA.Weee.Requests.AatfReturn;
+    using EA.Weee.Requests.Organisations;
+    using EA.Weee.Web.Areas.AatfReturn.ViewModels;
+    using EA.Weee.Web.Infrastructure;
+    using Prsd.Core.Mapper;
+    using Services;
+    using Services.Caching;
+    using System;
+    using System.Threading.Tasks;
+    using System.Web.Mvc;
+    using Web.Controllers.Base;
+    public class ReceivedPCSListController : ExternalSiteController
+    {
+        private readonly Func<IWeeeClient> apiClient;
+        private readonly IWeeeCache cache;
+        private readonly BreadcrumbService breadcrumb;
+        private readonly IMapper mapper;
+
+        public ReceivedPCSListController(Func<IWeeeClient> apiClient,
+            IWeeeCache cache,
+            BreadcrumbService breadcrumb,
+            IMapper mapper)
+        {
+            this.apiClient = apiClient;
+            this.cache = cache;
+            this.breadcrumb = breadcrumb;
+            this.mapper = mapper;
+        }
+
+        // GET: AatfReturn/ReceivedPCSList
+        [HttpGet]
+        public async Task<ActionResult> Index(Guid organisationId, Guid returnId)
+        {
+            var viewModel = new ReceivedPCSListViewModel();
+
+            using (var client = apiClient())
+            {
+                var organisationName = (await client.SendAsync(User.GetAccessToken(), new GetOrganisationInfo(organisationId))).OrganisationName;
```

Comment:
We don't need to make an API call to get the Organisation name. Can use await cache.FetchOrganisationName(organisationId);